apache/superset
# fix(sqllab): Async queries are now fetched properly

### SUMMARY

This PR fixes a bug introduced by (https://github.com/apache/superset/pull/21186) that caused async queries to always render a refetch button on the first query fetch. This is because the first time an async query is run it doesn't have a prevQuery (previous query). Once the query is run (refetched through the button), there's an existing prevQuery so it would always work on the second click.


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

#### BEFORE:
![unsuccessfulAsyncFetch](https://user-images.githubusercontent.com/55605634/193934856-6379058d-d3fa-4b9c-a585-38def24cc5f6.gif)

#### AFTER:
![successfulAsyncFetch](https://user-images.githubusercontent.com/55605634/193934504-d7090249-0b80-4002-8fde-5b78eee3a552.gif)


### TESTING INSTRUCTIONS

- Ensure that you have a database with `Asynchronous query execution` enabled
- Go to SQL Lab and select a database, schema, and table schema
  - I used a Google Sheets db for testing locally
- Observe that the query runs without having to click "Fetch data preview" or "Refetch results" first.

Note: This also happens in the result set when you click "run", but that route is currently blocked. There is a fix available for this route here: https://github.com/apache/superset/pull/21667 - I pulled down this branch and tested, I can confirm that this also works for running queries and getting a full result (not just a table preview).

### ADDITIONAL INFORMATION
